File: javascript/leap/leap.js
export const isLeap = year => {
  var isLeap;
  if (divisible(year, 400)) {
    isLeap = true;
  } else if (divisible(year, 100)) {
    isLeap = false;
  } else if (divisible(year, 4)) {
    isLeap = true;
  } else {
    isLeap = false;
  }
  return isLeap;
};

const divisible = (dividend, divisor) => {
  return dividend % divisor == 0;
};

File: cpp/leap/leap.cpp
#include "leap.h"

// static function declarations
static bool divisible_by(int32_t year, int32_t divisor);

// function definitions
namespace leap {
    bool is_leap_year(int32_t year) {
        return divisible_by(year, 400) || (divisible_by(year, 4) && !divisible_by(year, 100));
    }
}  // namespace leap

bool divisible_by(int32_t year, int32_t divisor)
{
    return year % divisor == 0;
}
